staging/src/k8s.io/client-go/tools/leaderelection/resourcelock/interface.go

```diff
@@ -85,6 +95,9 @@ type Interface interface {
 	// Identity will return the locks Identity
 	Identity() string
 
+	// LeaderElectionConfig will return leader election's lease duration, lease renew and retry configurations
+	LeaderElectionConfig() (*int, *int, *int)
```

Comment:
odd to hide implementation behind an interface and then intentionally leaking this information. Who is a consumer of this? Isn't the consumer also the same who creates the object?

Reply:
Why can those values be nil?